platform/src/main/java/org/stellar/anchor/platform/service/TransactionService.java

```diff
@@ -301,6 +302,25 @@ void validateAsset(String fieldName, Amount amount) throws BadRequestException {
       throw new BadRequestException(
           String.format("'%s' is not a supported asset.", amount.getAsset()));
     }
+
+    List<AssetInfo> allAssets =
+        assets.stream()
+            .filter(assetInfo -> assetInfo.getAssetName().equals(amount.getAsset()))
+            .collect(Collectors.toList());
+
+    if (allAssets.size() == 1) {
+      AssetInfo targetAsset = allAssets.get(0);
+
+      if (targetAsset.getSignificantDecimals() != null) {
+        // Check that significant decimal is correct
+        if (decimal(amount.getAmount(), targetAsset).compareTo(decimal(amount.getAmount())) != 0) {
+          throw new BadRequestException(
+              String.format(
+                  "'%s' has invalid significant decimals. Expected: '%s'",
+                  amount.getAmount(), targetAsset.getSignificantDecimals()));
+        }
+      }
+    }
```

Comment:
I agree we should not let businesses use too many decimals like `10.001` when `significant_decimals` is `2`. But I think they should be able to use fewer decimals, for example sending `10` should work.

Reply:
It will work, there is a test for that